### ISSUE TITLE
Add ASWebAuthenticationSession.Callback support

### ISSUE THREAD

**Issue by kay-schecker** [ThexXTURBOXx/flutter_web_auth_2#104]
## Is your feature request related to a problem? Please describe.
If I wanna use the **https scheme** and [Universal Links](https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html) on iOS the completionHandler is not called.

- This is related to Issue #65 

## Describe the solution you'd like
Apple has [deprecated the previous initializer](https://developer.apple.com/documentation/authenticationservices/aswebauthenticationsession/2990952-init) matching in favor of the [new initializer](https://developer.apple.com/documentation/authenticationservices/aswebauthenticationsession/4316247-init) factory functions:

- `ASWebAuthenticationSession.Callback.customScheme(String)`
- `ASWebAuthenticationSession.Callback.https(host: String, path: String)`

**Comment by ThexXTURBOXx** [comment]
Sadly, my MacOS VM is not working properly anymore, so I cannot work on this issue without blindly replacing things. If you have access to Apple hardware (or a VM), feel free to contribute via PR! :)

**Comment by ThexXTURBOXx** [comment]
I finally had access to a Macbook of a friend of mine.
At least I was able to add partial support for the new initializer.
I tested it on a simulator for iOS and the macBook itself - both worked fine.
However, I don't know if this breaks older environments, so it is good to have this published within the 4.x alpha cycle.
For now, I have not added `ASWebAuthenticationSession.Callback.https(host: String, path: String)`, but I may do that at some other point.
Hence, I will keep this issue open for now.
